clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -113,13 +182,23 @@ public void commitSync(final Duration timeout) {
         try {
             commitFuture.get(timeout.toMillis(), TimeUnit.MILLISECONDS);
         } catch (final TimeoutException e) {
-            throw new org.apache.kafka.common.errors.TimeoutException("timeout");
+            throw new org.apache.kafka.common.errors.TimeoutException(
+                     "timeout");
         } catch (final Exception e) {
             // handle exception here
             throw new RuntimeException(e);
         }
     }
 
+    /**
+     * Get the current subscription.  or an empty set if no such call has
+     * been made.
+     * @return The set of topics currently subscribed to
+     */
+    public Set<String> subscription() {
```

Comment:
```suggestion
    Set<String> subscription() {
```

I don't think this method should be `public`, should it? Let's do what we can to stick to the `Consumer` API.

Reply:
I think it's public in the consumer API, og impl:
```/**
     * Get the current subscription. Will return the same topics used in the most recent call to
     * {@link #subscribe(Collection, ConsumerRebalanceListener)}, or an empty set if no such call has been made.
     * @return The set of topics currently subscribed to
     */
    public Set<String> subscription() {
   ...
    }```

Reply:
I just double-checked; this is already public. I'm surprised it actually compiled with a package-private method.

@philipnee , can we use the `@Override` annotation for the interface methods we're implementing? That'll help to document which methods are from the interface.